pkg/controller/configuration/pipeline.go

```diff
@@ -366,10 +366,13 @@ func (p *updatePipeline) UpdateConfigVersion(revision string) *updatePipeline {
 func (p *updatePipeline) Sync() *updatePipeline {
 	return p.Wrap(func() error {
 		if p.ConfigConstraintObj != nil && !p.isDone() {
-			if err := SyncEnvConfigmap(*p.configSpec, p.newCM, &p.ConfigConstraintObj.Spec, p.Client, p.Context); err != nil {
+			if err := SyncEnvConfigmap(*p.configSpec, p.newCM, &p.ConfigConstraintObj.Spec, p.Client, p.Context, p.ctx.Cluster, p.ctx.SynthesizedComponent); err != nil {
```

Comment:
AsSecret is only applicable to configuration templates that allow injectEnvTo. AsSecret is not supported for non-envFrom configuration templates. The reason is that the related modifications are too large. This may be used to support it in future.

For the configuration template of injectEnvTo, the rendered configmap/secret won't be used, so the configmap/secret of InjectEnvTo does not need to be created. Here ignore the secret for compatibility considerations, the previous ComponentTemplateSpec.VolumeName is a required field, addon may mount the volume to the container, and removing it directly may cause problems.